gloss/crt0.S

```diff
@@ -142,6 +142,10 @@ _start:
 
 _skip_init:
 
+  /* All harts initialize their local interrupt controller */
+  csrr a0, mhartid
```

Comment:
this passes mhartid, but riscv_cpu_intc_init receives intc. Probably best to just pass nothing and have intc_init figure things out itself.